src/rssitem.cpp

```diff
@@ -144,7 +144,15 @@ nonstd::optional<std::string> RssItem::attribute_value(const std::string&
 	} else if (attribname == "author") {
 		return utils::utf8_to_locale(author());
 	} else if (attribname == "content") {
-		return utils::utf8_to_locale(description());
+		if (description_.has_value()) {
+			return utils::utf8_to_locale(description_.value());
+		} else if (ch) {
+			ch->fetch_description(this);
+			auto result = utils::utf8_to_locale(description_.value());
+			description_.reset();
+			return result;
+		}
+		return "";
```

Comment:
It looks like we'll need a mutex around `description_` to prevent a couple of concurrency bugs here:

1. the first `if` exhibits "time-of-check time-of-use " bug, where `description_` might get reset in between calls to  `has_value()` and `value()`;

2. there's a race between the two `if`s: if two query feeds contain the same item *and* both feeds get reloaded simultaneously, they might both ask for the description at the same time.

Reply:
Also, it might be useful to put a `ScopeMeasure` here, to get an idea of how expensive those "cache misses" are.

Reply:
Both done, thanks for the review!